models/repo.go

```diff
@@ -205,10 +205,11 @@ type Repository struct {
 	ExternalMetas map[string]string `xorm:"-"`
 	Units         []*RepoUnit       `xorm:"-"`
 
-	IsFork   bool        `xorm:"INDEX NOT NULL DEFAULT false"`
-	ForkID   int64       `xorm:"INDEX"`
-	BaseRepo *Repository `xorm:"-"`
-	Size     int64       `xorm:"NOT NULL DEFAULT 0"`
+	IsFork         bool        `xorm:"INDEX NOT NULL DEFAULT false"`
+	ForkID         int64       `xorm:"INDEX"`
+	BaseRepo       *Repository `xorm:"-"`
+	Size           int64       `xorm:"NOT NULL DEFAULT 0"`
+	IndexCommitSha string
```

Comment:
Maybe a standalone table to store `RepoID`, `BranchName`, `LastCommitSHA`, so that we can support search for an extra branch.